airflow/jobs.py

```diff
@@ -137,6 +137,13 @@ def on_kill(self):
         """
         pass
 
+    def on_failure(self, e):
+        """
+        Will be called when an exception happens during _execute
+        default just raise exception again
+        """
+        raise e
```

Comment:
It is better for this to be
```suggestion
        raise
```

That way it will preserve the original stack trace. With it as `raise e` it will appear to come from this line which will make it harder to debug/track down problems.

(May not be relevant as we can possibly remove this addition entirely)